File: Linkpoint/src/main/java/com/linkpoint/slproto/messages/SubscribeLoad.java
package com.linkpoint.slproto.messages;

import com.linkpoint.slproto.SLMessage;
import java.nio.ByteBuffer;

public class SubscribeLoad extends SLMessage {
    public SubscribeLoad() {
        this.zeroCoded = false;
    }

    public int CalcPayloadSize() {
        return 4;
    }

    public void Handle(SLMessageHandler sLMessageHandler) {
        sLMessageHandler.HandleSubscribeLoad(this);
    }

    public void PackPayload(ByteBuffer byteBuffer) {
        byteBuffer.putShort(-1);
        byteBuffer.put((byte) 0);
        byteBuffer.put((byte) 7);
    }

    public void UnpackPayload(ByteBuffer byteBuffer) {
    }
}

File: Linkpoint/src/main/java/com/linkpoint/slproto/messages/SubscribeLoad.kt
package com.linkpoint.slproto.messages

import com.linkpoint.slproto.SLMessage
import java.nio.ByteBuffer

class SubscribeLoad : SLMessage() {
    init {
        zeroCoded = false
    }

    override fun CalcPayloadSize(): Int = 4

    override fun Handle(handler: SLMessageHandler) {
        handler.HandleSubscribeLoad(this)
    }

    override fun PackPayload(buffer: ByteBuffer) {
        buffer.putShort(-1)
        buffer.put(0.toByte())
        buffer.put(7.toByte())
    }

    override fun UnpackPayload(buffer: ByteBuffer) {}
}